src/components/Table/Table.tsx

```diff
@@ -0,0 +1,129 @@
+
+import { bemCls, getVariantClasses } from '../../utils';
+
+import { defaultSortingIcons } from './Table.const';
+import { TableProps } from './Table.types';
+
+const Table: FC<TableProps<object>> = ({
+	className,
+	onSortChange,
+	options,
+	pagination,
+	rootClassName: root = 'c-table',
+	sortingIcons = defaultSortingIcons,
+	style,
+	variants,
+}) => {
+	const bem = bemCls.bind(root);
+	const rootCls = clsx(className, root, getVariantClasses(root, variants));
+
+	// State
+
+	const data = useMemo(() => options.data, [options.data]);
+	const columns = useMemo(() => options.columns, [options.columns]);
+
+	const instance = useTable(
+		{ ...options, manualSortBy: true, disableMultiSort: true, columns, data },
+		useSortBy,
+		usePagination
+	);
+
+	// Destructuring
+
+	const {
+		getTableBodyProps,
+		getTableProps,
+		headerGroups,
+		page,
+		prepareRow,
+		state: { sortBy },
+	} = instance;
+
+	// Effects
+
+	useEffect(() => {
+		onSortChange && onSortChange(sortBy);
+	}, [sortBy, onSortChange]);
+
+	// Render
+
+	const renderSortingIndicator = (column: HeaderGroup) => {
+		if (!column.canSort || column.disableSortBy) return null;
+
+		if (column.isSorted) {
+			return column.isSortedDesc ? sortingIcons.desc : sortingIcons.asc;
+		}
+
+		return sortingIcons.default;
+	};
+
+	return (
+		<>
+			<div className={clsx(bem('scroller'))}>
+				<table {...getTableProps()} className={rootCls} style={style}>
+					<thead className={clsx(bem('wrapper'), bem('wrapper', 'header'))}>
+						{headerGroups.map((group, i) => (
+							<tr
+								{...group.getHeaderGroupProps()}
+								key={i}
+								className={clsx(bem('row'), bem('row', 'header'))}
+							>
+								{group.headers.map((column, j) => (
+									<th
+										{...column.getHeaderProps(column.getSortByToggleProps())}
+										className={clsx(
+											bem('cell'),
+											bem('cell', 'header'),
+											column.isSorted && bem('cell', 'active')
+										)}
+										key={`${i}-${j}`}
+									>
+										{column.render('Header')}
+
+										{renderSortingIndicator(column)}
+									</th>
+								))}
+							</tr>
+						))}
+					</thead>
+
+					<tbody
+						className={clsx(bem('wrapper'), bem('wrapper', 'body'))}
+						{...getTableBodyProps()}
+					>
+						{page.map((row, i) => {
+							prepareRow(row);
+
+							return (
+								<tr
+									className={clsx(bem('row'), bem('row', 'body'))}
+									{...row.getRowProps()}
+									key={i}
+								>
+									{row.cells.map((cell, j) => {
+										return (
+											<td
+												{...cell.getCellProps()}
+												className={clsx(bem('cell'), bem('cell', 'body'))}
+												key={`${i}-${j}`}
+											>
+												{cell.render('Cell')}
+											</td>
+										);
+									})}
+								</tr>
+							);
+						})}
+					</tbody>
+				</table>
+			</div>
+
+			<div className={clsx(bem('footer'))}>{pagination && pagination(instance)}</div>
```

Comment:
Gaan we de pagination toevoegen in de tabel, willen we deze niet juist so simpel mogelijk houden?

Reply:
de `pagination`-functie hier geeft enkel toegang tot de `react-table` pagination props. De rendering van de pagination zelf gebeurt nog steeds top-level. Zie: https://github.com/viaacode/hetarchief-client/pull/53/files#diff-8e895a93e3239f21c9099e3315825fdf14b3dd303cec96dc885e6753425cf2abR37-R48